### ISSUE TITLE
Bug in `tf.autodiff.ForwardAccumulator` for `tf.signal.mdct`

### ISSUE THREAD

**Issue by eeDigitalSeal** [tensorflow/tensorflow#56737]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.9

### Current Behaviour?

```shell
Gradient in forward mode is inconsistent with that in backward mode for API `tf.signal.mdct`
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
signals = tf.constant([3.3425903], dtype=tf.float32)
frame_length = 49152
with tf.GradientTape(persistent=True,) as g:
      g.watch(signals)
      res = tf.reduce_sum(tf.signal.mdct(signals, frame_length, window_fn=None, pad_end=True, norm=None))
      print(res)
print("gradient in reverse mode: ", g.gradient(res, signals))

with tf.autodiff.ForwardAccumulator(signals,tf.constant([1.],shape=[1],dtype=tf.float32)) as acc:
        res = tf.reduce_sum(tf.signal.mdct(signals, frame_length, window_fn=None, pad_end=True, norm=None, ))
        print(res)
print("gradient in forward mode: ", acc.jvp(res))
```

### Relevant log output

```shell
tf.Tensor(11.523682, shape=(), dtype=float32)
gradient in reverse mode:  tf.Tensor([1.0006392], shape=(1,), dtype=float32)
tf.Tensor(11.523682, shape=(), dtype=float32)
gradient in forward mode:  tf.Tensor(3.4477844, shape=(), dtype=float32)
```
</details>

**Comment by sushreebarsa** [comment]
@gowthamkpr I was able to replicate the issue on colab , please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/6978b0e86f8071c814a3c02900edb39d/56737.ipynb). Thank you!

**Comment by cantonios** [comment]
You're running into floating-point precision issues because the frame length is too large (representable numbers go out of range in forward accumulation of gradients).  Changing to float64 fixes the issue.
